aspnetcore/client-side/blazor/introduction/index.md

```diff
@@ -0,0 +1,170 @@
+
+Interpreted mode is similar to how the core internals of the desktop Common Language Runtime (CLR) are distributed precompiled into native code, which then loads and executes .NET assembly files. One key difference is that the desktop CLR uses just-in-time (JIT) compilation extensively to make execution faster, while Mono on WebAssembly is closer to a pure interpretation model.
+
+### Ahead-of-time (AOT) compilation mode
+
+In ahead-of-time (AOT) compilation mode, the app's .NET assemblies are transformed into WebAssembly binaries at build time. At runtime, there's no interpretation. The code executes immediately. It's still necessary to load part of the Mono runtime. Features for low-level .NET services are loaded, such as garbage collection. Features for parsing .NET assemblies aren't required and aren't loaded.
+
+![AOT mode](index/_static/aotmode.png)
+
+This is similar to how the ngen tool has historically allowed AOT compilation of .NET binaries into native machine code. More recently, CoreRT provides a complete native AOT .NET runtime for executing .NET binaries.
+
+## Interpreted versus AOT
+
+Interpreted mode provides a faster development cycle than AOT. When code is changed under the interpreted model, the app can be rebuilt and reloaded in the browser within a few seconds. An AOT rebuild might take several minutes to compile and reload. Because interpreted mode offers faster recompilation, it might become the default mode for compilation during development. More work is required before a conclusion can be drawn on how to best apply interpreted and AOT modes of compilation with WebAssembly.
+
+## Blazor, a SPA framework
+
+Blazor has a growing standard feature set to solve common app requirements, such as UI composition, state management, and routing. Features are designed around the strengths of .NET and the C# language with careful consideration given to tooling support.
+
+Blazor is inspired by today's top SPA frameworks such as React, Vue, and Angular, plus some Microsoft technologies, such as Razor Pages. The goal is to provide SPA features that web developers have found most successful in existing frameworks in a way that capitalizes on .NET's strengths.
+
+## Blazor components
+
+In SPA frameworks, apps are built with *components*. A component usually represents a piece of UI, such as a page, dialog, or data entry form. Components can be nested, reused, and shared between projects.
+
+In Blazor, a component is a .NET class. The class can either be written directly, as a C# class (*\*.cs*), or more commonly in the form of a Razor markup page (*\*.cshtml*).
+
+Razor, which has been around since 2010, is a syntax for combining HTML markup with C# code. Razor is designed for developer productivity, allowing the developer to switch between markup and C# in the same file with IntelliSense support. Here's an example of how to express a simple custom dialog component in a Razor file:
```

Comment:
Razor, ~which has been around since 2010,~ is a syntax for combining HTML markup with C# code.
link to https://docs.microsoft.com/en-us/aspnet/core/mvc/views/razor